### ISSUE TITLE
1.9 Final Steps!!

### ISSUE THREAD

**Issue by DanVanAtta** [triplea-game/triplea#1196]
### fixes
- [x] Fix route drawer horizontal line bug (https://github.com/triplea-game/triplea/issues/1056) -  fix is in: https://github.com/triplea-game/triplea/pull/1199
- [x] Fix Mapdownload NPE - reported in: , http://tripleadev.1671093.n2.nabble.com/TripleA-1-9-0-0-Prerelease-tp7592899p7593479.html fixed in: https://github.com/triplea-game/triplea/pull/1194
- [x] Fix Map Name mismatch on load when loading master.zip files. - Fixed in: https://github.com/triplea-game/triplea/pull/1200
- [x] Export map  'm_' property not found  reflection error (java.lang.NoSuchFieldException: m_constantProperties, reported below in this issue) - fix is in https://github.com/triplea-game/triplea/pull/1201
- [x] Fix: https://github.com/triplea-maps/middle_earth-map_skin2/issues/2, map skins are broken. Fixed in: https://github.com/triplea-game/triplea/pull/1204
- [x] Fix missing error handling in substance look and feel from killing the game: https://github.com/triplea-game/triplea/issues/1208; mitigated in: https://github.com/triplea-game/triplea/pull/1211
- [x] Investigate and fix intermittent failure to download (http://tripleadev.1671093.n2.nabble.com/TripleA-1-9-0-0-Prerelease-tp7592899p7593506.html)
  Fixed/mitigated in: https://github.com/triplea-game/triplea/pull/1210
---
### lobby deployment
- [x] build new candidate (1.9.0.0.3194)
- [x] deploy to lobby (scripts are in: https://github.com/triplea-game/lobby)
- [x] deploy bots (scripts are in: https://github.com/triplea-game/lobby)
- [x] download latest 1.9 maps, get it working with bots (map download script is also in: https://github.com/triplea-game/lobby, required engine patch to get bots working: https://github.com/triplea-game/triplea/pull/1225)
---
### Late Stage Bug Fixes
- [x] Deal with map mods
- [x] Frankie's bots are getting remote not found exception
- [x] Broken open folders button: https://github.com/triplea-game/triplea/issues/1234
---
### testing
- [x] Check lobby compatibility with latest point release
- [x] Finalize lobby testing
  - [x] lobby works
  - [x] bots work
---
### more testing
- [x] check map images
- [x] skins and variants work
- [x] check PBEM
- [x] click every menu option, they should all work.
- [x] map downloads all working
---
- [x] Update triplea-game.org download link (v3266)
- [x] Update SourceForge by removing old pages and referencing new site and warclub
- [x] Update old lobby message to indicate 1.9 is released
- [x] Update SVN latest lobby version file
- [x] Update WarClub forums with a post
- [x] Update SourceForge Forums with a Post
- [x] Update A&A.org forums with a Post
- [ ] Update partner sites

Some ongoing, relevant tasks:
- Update any out of date information: https://github.com/triplea-game/triplea/blob/master/docs/admin/how_to_release.md

**Comment by Cernelius** [comment]
I believe you should add updating the "Who's Online" block of the WarClub; I mean the one that reads:

> Get Started!
> To play some great strategy games, follow these steps!
> Download TripleA from SourceForge
> We recommend the "all_platforms" zip file.
> Install some maps
> There are classic games and lots of mods!

The WarClub is the first (if not foremost) referring point for the gaming community, so it should not send you in the wrong place, to download TripleA.

Side note, even if I guess we don't "recommend the "all_platforms" zip file" anymore, apparently, I still believe the TripleA.exe should be inside.

Can ZjelcoP or somebody do the WarClub download link update?

Also, since there have been some changes, lately, it might be advisable not rushing too much, since I think the prerelease gets very little testing.

**Comment by Cernelius** [comment]
I think that the 3074 is not a good candidate.

The Napoleonic Empires mapskin doesn't work anymore (for me, at least).

You get View/Map Skins/master

And if you click on "master", you get this error:

> triplea.engine.version.bin:1.9
> Loading map: napoleonic_empire, from: C:\Users\Totale\triplea\downloadedMaps\napoleonic_empire-master.zip
> Loading map: napoleonic_empire, from: C:\Users\Totale\triplea\downloadedMaps\napoleonic_empire-master.zip
> Error Changing Map Skin2Loading map: napoleonic_empire-master, from: C:\Users\Totale\triplea\downloadedMaps\napoleonic_empire-master.zip
> 
> java.lang.NullPointerException
> 	at games.strategy.util.PointFileReaderWriter.readOneToManyPolygons(PointFileReaderWriter.java:207)
> 	at games.strategy.triplea.ui.mapdata.MapData.<init>(MapData.java:153)
> 	at games.strategy.triplea.ui.UIContext.internalSetMapDir(UIContext.java:77)
> 	at games.strategy.triplea.ui.AbstractUIContext.setMapDir(AbstractUIContext.java:122)
> 	at games.strategy.triplea.ui.TripleAFrame.updateMap(TripleAFrame.java:2082)
> 	at games.strategy.triplea.ui.menubar.ViewMenu.lambda$addMapSkinsMenu$667(ViewMenu.java:295)
> 	at games.strategy.triplea.ui.menubar.ViewMenu$$Lambda$80/1302735333.actionPerformed(Unknown Source)
> 	at javax.swing.AbstractButton.fireActionPerformed(Unknown Source)
> 	at javax.swing.AbstractButton$Handler.actionPerformed(Unknown Source)
> 	at javax.swing.DefaultButtonModel.fireActionPerformed(Unknown Source)
> 	at javax.swing.JToggleButton$ToggleButtonModel.setPressed(Unknown Source)
> 	at javax.swing.AbstractButton.doClick(Unknown Source)
> 	at javax.swing.plaf.basic.BasicMenuItemUI.doClick(Unknown Source)
> 	at javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(Unknown Source)
> 	at java.awt.AWTEventMulticaster.mouseReleased(Unknown Source)
> 	at java.awt.Component.processMouseEvent(Unknown Source)
> 	at javax.swing.JComponent.processMouseEvent(Unknown Source)
> 	at java.awt.Component.processEvent(Unknown Source)
> 	at java.awt.Container.processEvent(Unknown Source)
> 	at java.awt.Component.dispatchEventImpl(Unknown Source)
> 	at java.awt.Container.dispatchEventImpl(Unknown Source)
> 	at java.awt.Component.dispatchEvent(Unknown Source)
> 	at java.awt.LightweightDispatcher.retargetMouseEvent(Unknown Source)
> 	at java.awt.LightweightDispatcher.processMouseEvent(Unknown Source)
> 	at java.awt.LightweightDispatcher.dispatchEvent(Unknown Source)
> 	at java.awt.Container.dispatchEventImpl(Unknown Source)
> 	at java.awt.Window.dispatchEventImpl(Unknown Source)
> 	at java.awt.Component.dispatchEvent(Unknown Source)
> 	at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
> 	at java.awt.EventQueue.access$500(Unknown Source)
> 	at java.awt.EventQueue$3.run(Unknown Source)
> 	at java.awt.EventQueue$3.run(Unknown Source)
> 	at java.security.AccessController.doPrivileged(Native Method)
> 	at java.security.ProtectionDomain$1.doIntersectionPrivilege(Unknown Source)
> 	at java.security.ProtectionDomain$1.doIntersectionPrivilege(Unknown Source)
> 	at java.awt.EventQueue$4.run(Unknown Source)
> 	at java.awt.EventQueue$4.run(Unknown Source)
> 	at java.security.AccessController.doPrivileged(Native Method)
> 	at java.security.ProtectionDomain$1.doIntersectionPrivilege(Unknown Source)
> 	at java.awt.EventQueue.dispatchEvent(Unknown Source)
> 	at games.strategy.engine.framework.GameRunner$1.dispatchEvent(GameRunner.java:353)
> 	at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
> 	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
> 	at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
> 	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
> 	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
> 	at java.awt.EventDispatchThread.run(Unknown Source)


But this is what I was talking about. If I didn't make this test, you would have probably released 1.9 with this major bug. Since the prerelease gets very little testing, I think it needs staying there a long time with no changes, before making it into a release. Alternatively, doing the two steps "unstable" then "stable" release, as Veqryn used to do in the past.

**Comment by simon33-2** [comment]
Does this still have the issue that it can't open 1.8.0.9 save games? Perhaps a script to convert xml exports to 1.9 would be a fix?

**Comment by Cernelius** [comment]
Just tested, and it successfully opened a 1.8.0.9 savegame I recently took.

Exporting savegames is very important, but not for normal user; so, as long it gets made in 1 months or so from now, I think it doesn't need to be in the release.

Just tested, exporting the xml, and I still receive this error:

> triplea.engine.version.bin:1.9
> Loading map: napoleonic_empire, from: C:\Users\Totale\triplea\downloadedMaps\napoleonic_empire-master.zip
> Loading map: napoleonic_empire, from: C:\Users\Totale\triplea\downloadedMaps\napoleonic_empire-master.zip
> An Error occured whilst trying trying to setup the Property List
> java.lang.NoSuchFieldException: m_constantProperties
> 	at java.lang.Class.getDeclaredField(Unknown Source)
> 	at games.strategy.engine.data.export.GameDataExporter.propertyList(GameDataExporter.java:149)
> 	at games.strategy.engine.data.export.GameDataExporter.<init>(GameDataExporter.java:67)
> 	at games.strategy.triplea.ui.menubar.ExportMenu.exportXMLFile(ExportMenu.java:115)
> 	at games.strategy.triplea.ui.menubar.ExportMenu.lambda$addExportXML$608(ExportMenu.java:88)
> 	at games.strategy.triplea.ui.menubar.ExportMenu$$Lambda$109/1374933758.accept(Unknown Source)
> 	at games.strategy.ui.SwingAction$1.actionPerformed(SwingAction.java:42)
> 	at javax.swing.AbstractButton.fireActionPerformed(Unknown Source)
> 	at javax.swing.AbstractButton$Handler.actionPerformed(Unknown Source)
> 	at javax.swing.DefaultButtonModel.fireActionPerformed(Unknown Source)
> 	at javax.swing.DefaultButtonModel.setPressed(Unknown Source)
> 	at javax.swing.AbstractButton.doClick(Unknown Source)
> 	at javax.swing.plaf.basic.BasicMenuItemUI.doClick(Unknown Source)
> 	at javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(Unknown Source)
> 	at java.awt.AWTEventMulticaster.mouseReleased(Unknown Source)
> 	at java.awt.Component.processMouseEvent(Unknown Source)
> 	at javax.swing.JComponent.processMouseEvent(Unknown Source)
> 	at java.awt.Component.processEvent(Unknown Source)
> 	at java.awt.Container.processEvent(Unknown Source)
> 	at java.awt.Component.dispatchEventImpl(Unknown Source)
> 	at java.awt.Container.dispatchEventImpl(Unknown Source)
> 	at java.awt.Component.dispatchEvent(Unknown Source)
> 	at java.awt.LightweightDispatcher.retargetMouseEvent(Unknown Source)
> 	at java.awt.LightweightDispatcher.processMouseEvent(Unknown Source)
> 	at java.awt.LightweightDispatcher.dispatchEvent(Unknown Source)
> 	at java.awt.Container.dispatchEventImpl(Unknown Source)
> 	at java.awt.Window.dispatchEventImpl(Unknown Source)
> 	at java.awt.Component.dispatchEvent(Unknown Source)
> 	at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
> 	at java.awt.EventQueue.access$500(Unknown Source)
> 	at java.awt.EventQueue$3.run(Unknown Source)
> 	at java.awt.EventQueue$3.run(Unknown Source)
> 	at java.security.AccessController.doPrivileged(Native Method)
> 	at java.security.ProtectionDomain$1.doIntersectionPrivilege(Unknown Source)
> 	at java.security.ProtectionDomain$1.doIntersectionPrivilege(Unknown Source)
> 	at java.awt.EventQueue$4.run(Unknown Source)
> 	at java.awt.EventQueue$4.run(Unknown Source)
> 	at java.security.AccessController.doPrivileged(Native Method)
> 	at java.security.ProtectionDomain$1.doIntersectionPrivilege(Unknown Source)
> 	at java.awt.EventQueue.dispatchEvent(Unknown Source)
> 	at games.strategy.engine.framework.GameRunner$1.dispatchEvent(GameRunner.java:353)
> 	at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
> 	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
> 	at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
> 	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
> 	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
> 	at java.awt.EventDispatchThread.run(Unknown Source)

**Comment by simon33-2** [comment]
I've got a fix for that problem.

**Comment by simon33-2** [comment]
Just hasn't been merged.

**Comment by simon33-2** [comment]
That's not what I was suggesting in the other thread though. AFAIK the mechanism to load previous versions' save games doesn't work. What I was saying was to export the games as .xml files, update the xml for the changes in 1.9 and then load them into 1.9 as a work around.

**Comment by panther2** [comment]
Middle earth map skin does not work, either.
https://github.com/triplea-maps/middle_earth-map_skin2/issues/2

**Comment by prastle** [comment]
Tribes gave me an error

**Comment by panther2** [comment]
Looks like all skins are affected, I just tested world_war_ii_global-battlemap_skin, same error.

**Comment by panther2** [comment]
> Tribes gave me an error

Running fine on my 3074 with a very fresh map installation.

**Comment by DanVanAtta** [comment]
Updated the overview, we need to fix the map skins NPE. I think it is because there is no version field, and that is causing an error in code that assumes it. The route drawer I'd like us to take a crack at. Both seem like they are going to be pretty small patches. 

Kindly test anything else that can be. Might not be till tomorrow night that I can look into some of the latest issues.

**Comment by Cernelius** [comment]
> That's not what I was suggesting in the other thread though. AFAIK the mechanism to load previous versions' save games doesn't work. What I was saying was to export the games as .xml files, update the xml for the changes in 1.9 and then load them into 1.9 as a work around.

That sounds very power user. I'm the first one that would like to see the xml exporter working fully asap, but, since it has no direct value to normal users, I don't think the developers can be demanded to have it working for the release (of course, would be very nice, especially since it is a regression and it is very important to mapmakers).

**Comment by simon33-2** [comment]
Certainly is! But if step by step is provided, I reckon it could work.

**Comment by DanVanAtta** [comment]
@triplea-game/committers Took some liberties to get the latest 1.9 bug patches / non-controversial changes merged:
https://github.com/triplea-game/triplea/pull/1199
https://github.com/triplea-game/triplea/pull/1200
https://github.com/triplea-game/triplea/pull/1201
https://github.com/triplea-game/triplea/pull/1202
https://github.com/triplea-game/triplea/pull/1203 

Please feel free to leave any CR commentary, will be happy to feed more updates in as appropriate. Should have a new release candidate after these are merged (and hence the liberties).

**Comment by Cernelius** [comment]
Can @ron-murhammer update the "Who's Online" block that it is sending you to SourceForge to downlad TripleA, if ZjelcoP is not available?
I can really see that causing a lot of troubles, with people clicking on that...

**Comment by panther2** [comment]
Middle Earth and Global-battlemap skin are now working.
Napoleonic Empire-map-skin still not working.
https://github.com/triplea-maps/napoleonic_empire-political_map_skin/issues/2

**Comment by panther2** [comment]
Napoleonic Empire map skin is working fine now (3106). Thanks.

**Comment by ron-murhammer** [comment]
@Cernelius Yeah, once we officially release then I'll make sure warclub is updated to point to the new download. We also will be retiring the old SF page and just linking to the new tripleA and warclub sites.

@DanVanAtta Looks like there may still be some map download issues: http://tripleadev.1671093.n2.nabble.com/TripleA-1-9-0-0-Prerelease-tp7592899p7593506.html

**Comment by DanVanAtta** [comment]
I spotted those here and there. 'retry' fixes it. I'm not sure if there is anything to really be done in the short term. The existing code infrastructure is limiting, I suspect we are sending requests too quickly. It's worth a quick look though..

**Comment by DanVanAtta** [comment]
Updated issue overview with latest. Currently we're at a test phase, to try out as many different game features as possible, couple of outstanding problems have been identified (so no longer have a release candidate version).

**Comment by Cernelius** [comment]
If @simon33-2 really has a fix for the export xml, why not look into it? It is useless to normal users, thus no need to have it fixed for the release, but very important for mapmakers (and it might be an issue for new mapmakers, even after solved in future prereleases).
Just saying that if it is not too much of a fuss, I believe it should be worthwhile.

**Comment by DanVanAtta** [comment]
Please double check the latest build, I believe export.xml was fixed in: https://github.com/triplea-game/triplea/pull/1201

**Comment by simon33-2** [comment]
@Cernelius I did but it looks like there is a different fix. I've had that fix in my currently open PR for days or weeks.

Seems that the people with write access have a religious objection to merging anything which isn't perfection itself. I had started working on the sub combat casualty order limitations but (a) the code is awful (b) even if I fixed it I don't reckon it would be merged (c) I don't have much knowledge of Java and (d) Java is possibly the worst language ever devised. So I dropped it.

There will always be some issue used to justify not merging.

**Comment by DanVanAtta** [comment]
@simon33-2 you have not followed CR guidelines:
 - https://github.com/triplea-game/triplea/blob/master/docs/dev/code_guidelines.md
 - https://github.com/triplea-game/triplea/blob/master/docs/dev/code_format.md
You have not addressed all CR commentary, and frankly have been making excuses. 

I did not spot the fix in your PR. Splitting independent changes to their own PR will allow them to be considered individually and not get blocked by unrelated problems.

**Comment by simon33-2** [comment]
AFAIA, the only bit of feedback I hadn't addressed was the test, not counting feedback you've added today. I'm not really sure I know how to do that but when I feel motivated to pick this up again, I'll see if I can do it. Besides, what is the harm if this goes into the engine? Only maps which choose to opt into it will use it. The potential for damage is minimal and if there were a problem they could then opt out of it.

What is the CR guideline I haven't followed?

Regarding the fix for the export. I was pretty sure I mentioned it in a comment as well but I can't seem to find that. Sorry if I neglected to do that, as seems to be the case. Is it allowed to have multiple open PRs? I didn't know that - but each PR needs its own branch to draw from, is that correct?

**Comment by DanVanAtta** [comment]
@simon33-2 , will respond in PR https://github.com/triplea-game/triplea/pull/1140. Let's refocus this issue on the 1.9 release

**Comment by DanVanAtta** [comment]
1.9 bots patched, running with maps. Now purely waiting to see how testing goes :computer:

**Comment by DanVanAtta** [comment]
Looks like it may be time to pin the release version. We can likely fix anything not yet discovered in a point release.

**Comment by simon33-2** [comment]
Is there much chance of needing to use an old release to continue a game from here?

**Comment by prastle** [comment]
I have been trying to get players to use it but it seems that without forcing them to we need some kind of push!

**Comment by simon33-2** [comment]
Without backwards compatibility it's a bit of an imposition.

**Comment by DanVanAtta** [comment]
1.9 games can launch 1.8.0.9 games via the 'old jar' mechanism. The only trick from there is preserving 1.8 maps: http://www.triplea-game.org/upgrade_notes/

**Comment by panther2** [comment]
Is 1.7.0.5 really still needed? Are there really games going on?

**Comment by DanVanAtta** [comment]
@panther2 not really (https://github.com/triplea-game/triplea/pull/819), but perhaps for in case there is a really old save games that are being loaded

**Comment by DanVanAtta** [comment]
@ron-murhammer , are you up for taking care of the SVN pieces? Lobby message and home page updates. 

1.9 is looking ready, let's start releasing it : )

**Comment by simon33-2** [comment]
All this 1.8.0.9 mechanism does is fire up 1.8.0.9. Is that right? Is that all it ever did?

**Comment by simon33-2** [comment]
Why is the download version num 3133 when the most recent release is num 3154?

**Comment by DanVanAtta** [comment]
DL version is controlled here: https://github.com/triplea-game/triplea-game.github.io/blob/master/_config.yml

**Comment by simon33-2** [comment]
Ok, so are you asking me to update that file? If we did,  would it be backwards and forwards compatible with 3133, ie would both players of a game need to upgrade if one does? Or is the reason it hasn't been done because you aren't sure about those questions?

**Comment by DanVanAtta** [comment]
@simon33-2  the policy on updating the DL link version is not really set or well broken in.  It's updated to the latest release candidate now (https://github.com/triplea-game/triplea-game.github.io/pull/116)

The intent is not to keep pace with the bleeding edge, but perhaps close to it (provided we are not planning a major release).

**Comment by simon33-2** [comment]
I reckon bleeding edge is what we want in the prerelease stage. Let's find the problems ASAP.

**Comment by DanVanAtta** [comment]
@RoiEXLab pushed a bug fix, it's been deployed out to the lobby and bot reset too! 
3160 looks to be the first major. I've done some poking and prodding to check things out, PBEM etc.. all seem to work. Final verifications welcome. Looks like we should soon start trying to migrate people. Will be really nice getting rid of that ant script : )

**Comment by ron-murhammer** [comment]
It appears the one remaining issue now is variants: #1246 and #1247

**Comment by DanVanAtta** [comment]
Last items have been squared away, did a final smoke check, website updated to download the latest. Any objections if we start the migration to  1.9?

**Comment by simon33-2** [comment]
No objections from me.

**Comment by ron-murhammer** [comment]
@DanVanAtta I think its time. I should have some availability tomorrow to help release and update things.

**Comment by Cernelius** [comment]
- ww2v3_variants is still a broken collection of external mods.

https://github.com/triplea-game/triplea/issues/1247

- I still believe the all platform should come with the TripleA.exe inside (easy to paste it inside, but most people won't realise).

- It would be advisable the mapskins get debugged for the release, especially since it is a regression (you can have only 1 mapskin per map, now, the naming problem is the minor issue).

https://github.com/triplea-game/triplea/issues/1249

- The dowload window appears again behind the the main one, which is annoying.

https://github.com/triplea-game/triplea/issues/1260

My opinion of things I suggest tackling before releasing.

**Comment by Cernelius** [comment]
I wish to be able to write the map updating instructions before the release, but I can't right now.

**Comment by DanVanAtta** [comment]
Pretty good day doing lobby testing. We have a problem to fix though: https://github.com/triplea-game/triplea/issues/1262

Most other of the main use cases really look to have been covered, some folks were playing a game for a few rounds, found that bots could not load save games and that has been fixed: https://github.com/triplea-game/triplea/pull/1265

**Comment by DanVanAtta** [comment]
Courtesy @RoiEXLab #1262 is fixed.

Let's double check the upgrade/release instructions, and then make it official.

**Comment by DanVanAtta** [comment]
Website is updated to have 3266 (latest) as the download version.

**Comment by Cernelius** [comment]
@ron-murhammer did you get my 2 mails?? WAW40 should better be updated before the release, when most people will start downloading world_at_war. I also told in the dev forum.

**Comment by ron-murhammer** [comment]
@Cernelius Yeah, I just updated WAW40 with the fixes. The base GIMP image is too large to add to github. We still need to have further discussion on if/where we want to store those (preferably not in this thread).

**Comment by Cernelius** [comment]
I've made two Wiki pages with the information for upgrading and downgrading maps for this release.

https://github.com/triplea-game/triplea/issues/1190

I need somebody to reorder the pages, as I can't see how to do it; I suggest this order:

**Home
Broken Maps
Upgrade Maps Information
Downgrade Maps Information
Feature Back Log**

And I suggest you link this in developer forum, in a stickyed topic:

https://github.com/triplea-game/triplea/wiki/Upgrade-Maps-Information

**Comment by Cernelius** [comment]
This (what I see) is wrong in the chat room of the new one:

`Hosting: http://tinyurl.com/tripleaHosting\nLadder/Forum: http://tripleawarclub.org/`

Looks like you are missing an end line.

Also, isn't just "Welcome to TripleA 1.9" too generic? I mean shouldn’t it specify more and change at each release, instead of remaining the same for all 1.9 ones?

Also, I would say "in a private host", instead of "in direct host matches". Also, if you really want to keep the "old" jars, shouldn't you tell people that they will crash if they want to continue stuff like NWO or BW savegames, unless they move the maps to the user's folder? I believe most reloaded savegames will crash, and it will be quite a mess, since also Revised and Global players probably didn't download, as they were preinstalled.

**Comment by DanVanAtta** [comment]
I'm okay with  "Welcome to TripleA 1.9" being generic. Main point is that you are not in the 1.8 lobby. We can update later as needed. Meanwhile upgrading less often is a good thing. Lobby fixes probably should be tracked in a different place.

The items to discuss in this thread are the release / launch items.

@ron-murhammer , the ball is in your court for most of the rest of the work here. Veqryn intended to give me SVN creds but did not get around to it. Assignign it to you.



> Also, if you really want to keep the "old" jars, shouldn't you tell people that they will crash if they want to continue stuff like NWO or BW savegames, unless they move the maps to the user's folder?

You had said that nobody would read such instructions. If you want to take a crack at that type of update, please feel free: https://github.com/triplea-game/triplea-game.github.io/blob/master/upgrade_notes.md

**Comment by DanVanAtta** [comment]
@ron-murhammer , the ball is in your court for the rest of the work here. Veqryn intended to give me SVN creds but did not get around to it. Assigning it to you.

**Comment by Cernelius** [comment]
I think the instructions at the Website there are fine as they are, as also I don't think most will read them. I was thinking to specify in lobby that the backward compatibility of 1.9 is only for the maps that you installed, not the preinstalled ones, but likely still confusing to most people and might get too long (I mainly just suggested to remove the old jar for this release).

**Comment by ron-murhammer** [comment]
I will be going through the remaining release steps this weekend to get everyone moved over to 1.9 and begin retiring things related to SourceForge and 1.8.0.9. We will look to keep the 1.8.0.9 lobby open for a few weeks.

**Comment by Cernelius** [comment]
I know you have already released it, but can you reclassify it as 1.9.0.1 or whatever the like?

I believe it would be better keeping having a consistent release numbering, beside the prerelease one, mostly for mapmaking reference.

**Comment by Cernelius** [comment]
Also, this should be changed in the release notes, as the absence of the arrow was just a regression:

Change:

> Unit route drawing updated to draw smoother routes, with arrows at the end of the route line to indicate move direction.

To:

> Smoothed the route drawing.

The only other differences are that now the route is visible even before leaving the starting territory and that now it wraps for the Y axis too (you can see it in HexGlobe FFA), so it could be:

> Smoothed the route drawing and added Y axis wrapping support to it.

Also, what about fully detailing there how the new transport selection works, or at least mentioning it has been changed? That's a neat improvement, isn't it?

Regarding the puzzle games, instead of "inaccessible" you should say "already unsupported", as they have been already removed in the 1.8.0.9 engine, and the matter was just that the developers forgot to delete the relative preinstalled, but hidden, maps. Basically, having left them in the "maps" folder was just an oversight and a bug (you get the bug if you try to play them, as per instructions). Thus, I would reclassify the removal of the maps that were already removed for the 1.8.0.9 engine amongst the "Bug Fixes".

Also, I know you have already discussed and decided the matter, but I want to let you know that I believe having the name or names of the specific main developers and contributors beside the points was good.

**Comment by Cernelius** [comment]
Actually, I just checked, TicTacToe etc. were already not supported back in 1.8.0.7 (1.8.0.5 was the latest release you were able to play TicTacToe with):

> triplea.engine.version.bin:1.8.0.7
> Could not parse:file:/C:/Users/Totale/triplea/games/TripleA/triplea_1_8_0_7_all_platforms/triplea_1_8_0_7/maps/unclassified/games/tictactoe.xml
> games.strategy.engine.data.GameParseException: Class <games.puzzle.tictactoe.TicTacToe> could not be found.
> 	at games.strategy.engine.data.GameParser.getInstance(GameParser.java:455)
> 	at games.strategy.engine.data.GameParser.parseGameLoader(GameParser.java:558)
> 	at games.strategy.engine.data.GameParser.parse(GameParser.java:97)
> 	at games.strategy.engine.framework.ui.NewGameChooserEntry.<init>(NewGameChooserEntry.java:33)
> 	at games.strategy.engine.framework.ui.NewGameChooserModel.createEntry(NewGameChooserModel.java:167)
> 	at games.strategy.engine.framework.ui.NewGameChooserModel.addNewGameChooserEntry(NewGameChooserModel.java:140)
> 	at games.strategy.engine.framework.ui.NewGameChooserModel.populateFromDirectory(NewGameChooserModel.java:178)
> 	at games.strategy.engine.framework.ui.NewGameChooserModel.populate(NewGameChooserModel.java:77)
> 	at games.strategy.engine.framework.ui.NewGameChooserModel.<init>(NewGameChooserModel.java:33)
> 	at games.strategy.engine.framework.ui.NewGameChooser.refreshNewGameChooserModel(NewGameChooser.java:244)
> 	at games.strategy.engine.framework.startup.mc.GameSelectorModel$1.run(GameSelectorModel.java:365)
> 	at java.lang.Thread.run(Unknown Source)

**Comment by Cernelius** [comment]
I was thinking that the download buttons both in http://www.triplea-game.org/ and http://www.triplea-game.org/download/ don't specify the version.

Would be maybe better if they tell that you are downloading 1.9.0.0 (if this is the official full name of this release?), so people don't get confused thinking that is just the TripleA 1.8.0.9 they already have.

> 1.9.0.0 is now released and available for download at http://www.triplea-game.org/
> This 1.8.0.9 lobby will remain open for a few weeks to allow players to finish their 1.8.0.9 games then it will be closed
> Please start any new games with 1.9.0.0
> (03:08:05) AmericanKnight:  I can bring back a temp bot
> (03:08:09) AmericanKnight:  do you guys needone
> (03:08:21) Cernel:  hey everyone, better you move to the new lobby, the games started with this one won't be playable on new
> (03:08:36) elbert70:  how
> (03:08:42) Cernel:  there are already a bunch of people in the new one
> (03:08:44) Cernel:  uhm
> (03:08:46) elbert70:  is tht why no 1 is here?
> (03:08:52) Cernel:  kind of
> (03:08:56) Cernel:  currently split
> (03:09:06) Cernel:  but lack of bots might be a reason too
> Matthew959 has left
> (03:09:08) elbert70:  oh
> (03:09:08) Cernel:  anyway
> (03:09:13) Cernel:  download the new one
> (03:09:19) Cernel:  and go in the other lobby
> (03:09:26) Cernel:  this one is just for playing old games
> acheyne has left
> Geronimo_ has left
> elbert70 has left
> (03:09:37) Cernel:  like if you have a savegame
> Mountain_Breeze has left
> elbert70 has joined
> (03:09:59) AmericanKnight:  http://www.triplea-game.org/download/
> Matthew959 has joined
> Kaptain has left
> Geronimo_ has joined
> Geronimo_ has left
> (03:11:01) Matthew959:  how do you go to the new lobby?
> (03:12:15) AmericanKnight:  http://www.triplea-game.org/download/
> (03:13:04) Matthew959:  i upgraded to 1.90

**Comment by Cernelius** [comment]
> elbert70 has left
> elbert70 has joined
> elbert70 has left
> dqa77 has joined
> BammBamm75 has left
> EVERYBODYPANIC has left
> elbert70 has joined
> flak72 has joined
> turbodewd has joined
> flak72 has left
> (03:19:21) elbert70:  is this new lobby?
> (03:19:26) Cernel:  no!
> (03:19:27) Cernel:  lol
> warthunder has joined
> turbodewd has left
> (03:19:36) Cernel:  what did you do?
> (03:19:36) elbert70:  i dl wtf
> (03:19:47) AmericanKnight:  http://www.triplea-game.org/download/
> (03:19:47) Cernel:  uhm what did you download, the installer?
> (03:19:55) elbert70:  i think so
> (03:19:58) Cernel:  doesn't it overwrite?
> (03:20:09) elbert70:  i thought it wood
> (03:20:12) Cernel:  sorry I don't really know cause I don't use the installer; anyway, no this is the old lobby
> (03:20:16) elbert70:  u tell me
> (03:20:24) Cernel:  uhm
> (03:20:31) Cernel:  well
> (03:20:36) warthunder:  where are the bots?
> (03:20:37) Cernel:  now it should install beside the old one
> (03:20:42) Cernel:  thus no overwrite
> (03:20:49) Cernel:  but you should have both, I assume
> (03:21:11) elbert70:  ok i guess
> elbert70 has left

**Comment by Cernelius** [comment]
Also, just as a reference, the WOPR bots went down; just be warned that people will probably mix this issue up with the release, albeit being totally unrelated; just a notice:
http://www.tripleawarclub.org/modules/newbb/viewtopic.php?topic_id=2441

> (03:20:36) warthunder:  where are the bots?
> (03:20:37) Cernel:  now it should install beside the old one
> (03:20:42) Cernel:  thus no overwrite
> (03:20:49) Cernel:  but you should have both, I assume
> (03:21:11) elbert70:  ok i guess
> elbert70 has left
> hammond has left
> champ73 has joined
> (03:22:41) warthunder:  no bots tonight?
> champ73 has left
> (03:22:54) dqa77:  move to new lobby
> (03:23:04) dqa77:  new version is up
> SSRyan (1) has joined
> SSRyan (1) has left
> (03:23:48) warthunder:  how do you do all that?
> SSRyan has left
> flak72 has joined
> (03:25:38) Cernel:  it is explained at the start of the chat :O
> Matthew959 (1) has joined
> (03:25:50) Cernel:  1.9.0.0 is now released and available for download at http://www.triplea-game.org/
> AmericanKnight has left
> (03:26:30) Cernel:  also, new games are not backwards compatible, so you are really better off starting new in new (this one is for finishing old)
> dqa77 has left
> (03:28:03) Cernel:  no bots is another story, but there are a few in new

**Comment by Cernelius** [comment]
champ73 is a Mod and a very big veteran; I'm not sure what his problems are with this release; still trying to get people migrating:

> (03:30:19) champ73:  tried to dl the new lobby, it wants into my PC
> (03:30:32) champ73:  to heck with that
> (03:31:08) champ73:  not gonna allow "unknown" app into my PC
> (03:31:51) Cernel:  uh? it is just unknown because it is not famous stuff

**Comment by ron-murhammer** [comment]
@DanVanAtta The only thing that is complete here is updating partner sites which many of them I don't even have the login for. I think we should just close this and consider opening a new issue if we want to track down partner sites to update (along with get logins from Veqryn).
